src/vm/eventpipeconfiguration.cpp

```diff
@@ -256,8 +256,8 @@ EventPipeProvider *EventPipeConfiguration::GetProviderNoLock(const SString &prov
     CONTRACTL
     {
         THROWS;
-        GC_TRIGGERS;
-        MODE_PREEMPTIVE;
+        GC_NOTRIGGER;
+        MODE_ANY;
```

Comment:
As you are relaxing the contract, it would be nice to comment why we need to relax them?

Reply:
CI uncovered different path that triggered assertions that I did not want to handle them in this PR, therefore I reverted some of the contract changes.

Reply:
This is one of the assertions:
```
Assert failure(PID 8432 [0x000020f0], Thread: 8436 [0x20f4]): CONTRACT VIOLATION by EventPipeConfiguration::GetProvider at \"f:\\workspace\\_work\\1\\s\\src\\vm\\eventpipeconfiguration.cpp\" @ 245

GC_TRIGGERS encountered in a GC_NOTRIGGER scope

                        CONTRACT in EventPipeConfiguration::GetProvider at \"f:\\workspace\\_work\\1\\s\\src\\vm\\eventpipeconfiguration.cpp\" @ 245
VIOLATED-->  CONTRACT in EventPipe::GetProvider at \"f:\\workspace\\_work\\1\\s\\src\\vm\\eventpipe.cpp\" @ 524
                        CONTRACT in EventPipeInternal::GetProvider at \"f:\\workspace\\_work\\1\\s\\src\\vm\\eventpipeinternal.cpp\" @ 129
                        CONTRACT in DispatchCallSimple at \"f:\\workspace\\_work\\1\\s\\src\\vm\\callhelpers.cpp\" @ 185
                        CONTRACT in MethodTable::RunClassInitEx at \"f:\\workspace\\_work\\1\\s\\src\\vm\\methodtable.cpp\" @ 3331
                        GCX_COOP in MethodTable::DoRunClassInitThrowing at \"f:\\workspace\\_work\\1\\s\\src\\vm\\methodtable.cpp\" @ 3438
                        CONTRACT in MethodTable::DoRunClassInitThrowing at \"f:\\workspace\\_work\\1\\s\\src\\vm\\methodtable.cpp\" @ 3436
                        CONTRACT in MethodTable::CheckRunClassInitThrowing at \"f:\\workspace\\_work\\1\\s\\src\\vm\\methodtable.cpp\" @ 3735
                        CONTRACT in DynamicHelperFixup at \"f:\\workspace\\_work\\1\\s\\src\\vm\\prestub.cpp\" @ 2846
                        GCX_PREEMP_THREAD_EXISTS in DynamicHelperWorker at \"f:\\workspace\\_work\\1\\s\\src\\vm\\prestub.cpp\" @ 3209
                        CONTRACT in MethodDescCallSite::CallTargetWorker at \"f:\\workspace\\_work\\1\\s\\src\\vm\\callhelpers.cpp\" @ 344
                        GCX_COOP in Assembly::ExecuteMainMethod at \"f:\\workspace\\_work\\1\\s\\src\\vm\\assembly.cpp\" @ 1656
                        CONTRACT in Assembly::ExecuteMainMethod at \"f:\\workspace\\_work\\1\\s\\src\\vm\\assembly.cpp\" @ 1640
                        GCX_COOP in CorHost2::ExecuteAssembly at \"f:\\workspace\\_work\\1\\s\\src\\vm\\corhost.cpp\" @ 436
                        CONTRACT in CorHost2::ExecuteAssembly at \"f:\\workspace\\_work\\1\\s\\src\\vm\\corhost.cpp\" @ 385



CORECLR! <no symbol> + 0x0 (0x00007ff8`3a584646)
CORECLR! GetCLRRuntimeHost + 0x4F8B6 (0x00007ff8`3a65bca6)
CORECLR! CreateCLRProfiling + 0x24024 (0x00007ff8`3ad11a84)
CORECLR! GetCLRRuntimeHost + 0x36FE6E (0x00007ff8`3a97c25e)
CORECLR! CreateCLRProfiling + 0x694D0 (0x00007ff8`3ad56f30)
<no module>! <no symbol> + 0x0 (0x00007ff7`db46cf40)
<no module>! <no symbol> + 0x0 (0x00000098`0e97bc80)
<no module>! <no symbol> + 0x0 (0x00000098`0e97b9e0)
<no module>! <no symbol> + 0x0 (0x0000024c`09ee2c50)
<no module>! <no symbol> + 0x0 (0x00007ff7`db8f32f8)
    File: f:\\workspace\\_work\\1\\s\\src\\vm\\eventpipeconfiguration.cpp Line: 245
    Image: C:\\dotnetbuild\\work\\7abaa591-329d-4487-9783-ef582e2757a5\\Payload\\CoreRun.exe


Return code:      1
Raw output file:      C:\\dotnetbuild\\work\\7abaa591-329d-4487-9783-ef582e2757a5\\Work\\4db27c6d-4903-4693-b97f-a9e0ad004cc1\\Exec\\tracing\\eventcounter\\Reports\\tracing.eventcounter\\incrementingpollingcounter\\incrementingpollingcounter.output.txt
Raw output:
BEGIN EXECUTION
 \"C:\\dotnetbuild\\work\\7abaa591-329d-4487-9783-ef582e2757a5\\Payload\\corerun.exe\" incrementingpollingcounter.exe 
Test Passed
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=C:\\dotnetbuild\\work\\7abaa591-329d-4487-9783-ef582e2757a5\\Payload
> C:\\dotnetbuild\\work\\7abaa591-329d-4487-9783-ef582e2757a5\\Work\\4db27c6d-4903-4693-b97f-a9e0ad004cc1\\Exec\\tracing\\eventcounter\\incrementingpollingcounter\\incrementingpollingcounter.cmd
Expected: True
Actual:   False
```